### ISSUE TITLE
SetDefaultDllDirectories function missing parameter

### ISSUE THREAD

**Issue by diablodale** [MicrosoftDocs/feedback#3086]
at https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-setdefaultdlldirectories
is missing the `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` parameter.
I can see it mentioned in passing in the Remarks. However, it is not listed in the Parameters section.

### Expected

The parameters section to have a line entry describing `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR`

**Comment by kirill-21** [comment]
You sure it's missing? I think this is a wrong parameter. Specifying of this flag leads to 87 error, which is "The parameter is incorrect."

![image](https://github.com/MicrosoftDocs/feedback/assets/63228361/68e66293-cc84-4d4e-b248-7f28054866a6)

**Comment by diablodale** [comment]
Yes. The docs continue to not clearly document that flag.

My customers run this line of code on every app start, and no errors.
```c++
SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_DEFAULT_DIRS | LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR);
```

FYI: The screenshot you attached doesn't show anything relevant.

**Comment by diablodale** [comment]
You are collapsing two things together, so I will separate them

1. The OP documentation still live today contains a bullet with the text "The directory that contains the DLL (LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR). This directory is searched only for dependencies of the DLL being loaded." This bullet refers to "LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR" as a non-existing parameter. I am repeating what I already wrote in my OP. Either the doc's parameter is missing from the html table, or this bullet is errant text and should be removed.
2. Separately, you are inquiring if `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` causes a zero or non-zero to be returned from `SetDefaultDllDirectories()`. That inquiry is not this issue. This OP is an issue reporting a documentation problem. I don't have the free time to investigate this independent inquiry that you are posing. If you want to hire me to investigate your independent inquiry, you can buy hours at https://hidale.com/shop/consulting/

**Comment by kirill-21** [comment]
> You are collapsing two things together, so I will separate them
> 
> 1. The OP documentation still live today contains a bullet with the text "The directory that contains the DLL (LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR). This directory is searched only for dependencies of the DLL being loaded." This bullet refers to "LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR" as a non-existing parameter. I am repeating what I already wrote in my OP. Either the doc's parameter is missing from the html table, or this bullet is errant text and should be removed.
> 2. Separately, you are inquiring if `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` causes a zero or non-zero to be returned from `SetDefaultDllDirectories()`. That inquiry is not this issue. This OP is an issue reporting a documentation problem. I don't have the free time to investigate this independent inquiry that you are posing. If you want to hire me to investigate your independent inquiry, you can buy hours at https://hidale.com/shop/consulting/

There are not issues in the documentation, the `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` flag reffers to LoadLibraryEx method, which is described few sentences above that line you are talking about, you should read full description

**Comment by diablodale** [comment]
No, perhaps you are not a native English speaker. "LoadLibraryEx" does not appear "few sentences" above that line. It appears two paragraphs above the errant text and therefore is unrelated to the bullets. In English, paragraphs separate topics and thoughts. There is no connection between two paragraphs up "LoadLibraryEx" and the errant bullet.

If you disagree, I recommend you consult your senior editor.

**Comment by kirill-21** [comment]
> No, perhaps you are not a native English speaker. "LoadLibraryEx" does not appear "few sentences" above that line. It appears two paragraphs above the errant text and therefore is unrelated to the bullets. In English, paragraphs separate topics and thoughts. There is no connection between two paragraphs up "LoadLibraryEx" and the errant bullet.
> 
> If you disagree, I recommend you consult your senior editor.

Perhaps you have no idea what you are talking about and are not really familiar with Windows API.

![image](https://github.com/MicrosoftDocs/feedback/assets/63228361/fc913d8f-dd93-4910-a29b-2f7855e9acde)
You can not add this flag to SetDefaultDllDirectories because system won't understand which path you want to add, it specifies the relative path from the dll you're trying to load with LoadLibraryEx function.

The paragraph you are referring to is taken from the LoadLibraryEx wiki, it has nothing to do with SetDefaultDllDirectories
![image](https://github.com/MicrosoftDocs/feedback/assets/63228361/466691e0-4eb4-4afd-8d31-2d472ae12405)
 

Please, first spend at least a few minutes for research before arguing,

**Comment by diablodale** [comment]
I look forward to your more professional conversation. Meanwhile...

* Your top picture is irrelevant. Read my OP. I will not repeat my OP again
* Your 3rd paragraph "The paragraph you are referring to..." is false. Below is a screenshot (you love those) that proves my quote, "The directory that contains the DLL (LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR). This directory is searched only for dependencies of the DLL being loaded." exists on the `SetDefaultDllDirectories` function page. Both of these are surrounded with BIG RED LINES so you can see them.
* In addition, I put BIG RED CIRCLES around the linguistic and conceptual link of the `DirectoryFlags` which are listed in the errant bullets

![image](https://github.com/MicrosoftDocs/feedback/assets/679350/0904dbb7-b37c-4767-91b6-7b89d0465f34)

**Comment by kirill-21** [comment]
> * `SetDefaultDllDirectories`

I hope somebody from the microsoft team will explain you what's going on here sooner or later... Never seen such people, who can not really read what they are shown

**Comment by jffriend** [comment]
@kirill-21 A couple of your posts have been deleted for violation of our [code of conduct](https://github.com/MicrosoftDocs/feedback/blob/04fc1819f3bbf744dbf10b4d66bf93b8a9df27d1/CODE_OF_CONDUCT.md). Please be more open, friendly and patient in your interactions in this repo.

@diablodale, Sorry for the rudeness. I'm closing this discussion to avoid a repeat.
